Editing

sirpravdi opened the pull request:
code check up with linter + created dom helper

sirpravdi committed:
test2

sirpravdi committed:
code check up with linter + created dom helper

gohabereg commented on main.js:
```diff
@@ -1,57 +1,78 @@
+    }
+    /** processing click on emoji */
+    pollClick() {
+        var wrap = document.querySelector(".reactions-wrapper");
```
var стоит избегать. Последний стандарт ES определяет let и const

gohabereg commented on main.js:
```diff
@@ -1,57 +1,78 @@
-      localStorage.setItem('index' + this.parentElement.dataset.reactionsCounter, parseInt(localStorage.getItem('index' + this.parentElement.dataset.reactionsCounter)) + 1);
-      wrap.querySelector('[data-reactions-index="index' + this.parentElement.dataset.reactionsCounter + '"]').innerText = localStorage.getItem('index' + this.parentElement.dataset.reactionsCounter);
+    /** making creation of dom elements easier */
+    domHelper(elName, classList) {
```
Сюда можно добавить назначение атрибутов (вроде type, innerText, value и тд).
И метод лучше переименовать. domHelper не даёт понимания, что делает функция. Лучше назвать createElement или make, что-то подобное

sirpravdi committed:
Update main.js

replace Reactions,prototype by this, let wrap be const wrap

sirpravdi committed:
Update main.js

sirpravdi committed:
Update main.js

More specific localstorage keys
Replaced var with let/const
domHelper method was renamed into createElement
createElement method now can process attributes
Reactions constructor now can process poll parent element selector/id

sirpravdi committed:
Update main.js

sirpravdi committed:
Update main.js

sirpravdi committed:
Update main.js

gohabereg commented on main.js:
```diff
@@ -1,57 +1,85 @@
 class Reactions {
+    /**
+     * Create a reactions poll.
+     * @param {object} data - object containing poll emojis, title and parent element.
```
Можно описать внутренности `data`:
```
@param {object} data - ...
@param {HTMLElement} data.parent - ...
@param {string[]} data.reactions - ...
@param {string} title - ...
```

gohabereg replied:
Теперь у тебя `data.parent` не элемент, а селектор. Нужно поправить описание и тип параметра

gohabereg commented on main.js:
```diff
@@ -1,57 +1,85 @@
+        
+        wrap.append(pollTitle);
+        data.reactions.forEach((item, i, arr) => {
+            let counter = Reactions.prototype.createElement("div", "reactions-wrapper__counter");
```
Все вызовы через `prototype` нужно сделать через `this`

gohabereg replied:
И все переменные, которые не переназначаются нужно объявлять через `const`

gohabereg commented on main.js:
```diff
@@ -1,57 +1,85 @@
+            let counter = Reactions.prototype.createElement("div", "reactions-wrapper__counter");
+
+            wrap.append(counter);
+            let label = Reactions.prototype.createElement("label", "reactions-wrapper__emoji", {textContent: String.fromCodePoint(item)});
```
Подключи еще `eslint` с нашим конфигом, чтобы весь кодстайл был единым

sirpravdi replied:
Мне казалось, что я подключила, но, выходит, сделала как-то криво, т.к. почти весь код тут после линтера

gohabereg commented on main.js:
```diff
@@ -1,57 +1,85 @@
+    }
+    /** processing click on emoji */
+    pollClick() {
+        let wrap = document.querySelector(".reactions-wrapper");
```
Чтобы не искать каждый раз обертку, можно сохранить ее внутри класса

sirpravdi replied:
Я сделала, но не понимаю почему именно так работает в этом случае input.addEventListener("click", input => this.pollClick(input.target));
Почему тут input это не input, а click?

gohabereg commented on main.js:
```diff
@@ -1,57 +1,85 @@
-        index.dataset.reactionsIndex = 'index' + i;
-        if(!localStorage.getItem('index' + i)){
-          localStorage.setItem('index' + i, 0);
+        if (wrap.querySelector(".reactions-wrapper__emoji_picked")) {
```
Если `picked` это класс для выбранного эмодзи, то по БЭМ это модификатор — `reactions-wrapper__emoji--picked`

gohabereg commented on main.js:
```diff
@@ -1,57 +1,85 @@
 };
 
-new Reactions({title: 'What do yout think?', reactions: ['0x1F601','0x1F914','0x1F644']});
+new Reactions({parent: document.body, title: "What do you think?", reactions: ["0x1F601", "0x1F914", "0x1F644"]});
```
Удобнее, когда вместо элемента передается id или селектор, а модуль уже сам берет нужный элемент

gohabereg commented on main.js:
```diff
@@ -1,57 +1,85 @@
+            prev = wrap.querySelector(".reactions-wrapper__emoji_picked");
+
+            prev.classList.remove("reactions-wrapper__emoji_picked");
+            localStorage.setItem("reactionIndex" + prev.dataset.reactionsCounter, parseInt(localStorage.getItem("reactionIndex" + prev.dataset.reactionsCounter)) - 1);
```
Длинные строки, такие не очень удобно читать. Чтобы укоротить, сохраняй одинаковые части в переменные. Например тут можно сделать `storageKey = 'reactionIndex' + prev.dataset.reactionsCounter`

sirpravdi committed:
Update main.css

sirpravdi committed:
Update main.js

gohabereg commented on main.js:
```diff
@@ -1,57 +1,90 @@
+    constructor(data) {
+        this.wrap = this.createElement("div", "reactions-wrapper");
+        
+        document.querySelector(data.parent).append(this.wrap);
```
Тут обязательно нужна проверка на существование элемента. Если его нет, будет runtime error

gohabereg commented on main.js:
```diff
@@ -1,57 +1,90 @@
+        const pollTitle = this.createElement("span", "reactions-wrapper__title", {innerText: data.title});
+        
+        this.wrap.append(pollTitle);
+        data.reactions.forEach((item, i, arr) => {
```
Внутренности `forEach` можно вынести в отдельный метод, чтобы было что-то вроде `data.reactions.forEach((item, i) => this.addReaction(item, i))`

gohabereg commented on main.js:
```diff
@@ -1,57 +1,90 @@
+    pollClick(clicked) {
+        let prev = null;
+
+        if (this.wrap.querySelector(".reactions-wrapper__emoji--picked")) {
```
Тут можно упростить логику и уменьшить количество поисков по странице, если сохранять внутри модуля сами кнопки.

sirpravdi committed:
pass files from gitignore to rep

sirpravdi committed:
small codestyle fixes

gohabereg commented on .eslintrc.js:
```diff
@@ -0,0 +1,27 @@
+    "env": {
+        "browser": true
+    },
+    "extends": "codex",
```
Используй плиз только наш конфиг, без перегрузки правил.
У нас отступ 2 пробела, строки и одинарных ковычках

sirpravdi replied:
Конфиг постараюсь настроить как дома буду

gohabereg commented on package.json:
```diff
@@ -0,0 +1,19 @@
+  },
+  "author": "",
+  "license": "ISC",
+  "bugs": {
```
Чтобы использовать линтер, тебе нужно установить в зависимости наш конфиг и eslint.
Тут инструкция есть — https://github.com/codex-team/eslint-config

sirpravdi committed:
Update main.js

Created method addReaction - create and insert reactions button
Added error for case with invalid parent element
Picked reaction is now stored in module, pollClick method is changed according to that

sirpravdi committed:
Update main.js

throw fix

sirpravdi committed:
eslint setup

sirpravdi committed:
gitignore update

sirpravdi committed:
resolve conflicts

sirpravdi committed:
tem merge

sirpravdi committed:
resolve conflict

sirpravdi committed:
code checkup

sirpravdi committed:
Update main.js

pollClick was renamed into reactionClicked
added new methods - getCounter and setCounter (for localStorage access)
nextSibling and variables were addaed to make code more clear

gohabereg commented on main.js:
```diff
@@ -1,57 +1,142 @@
+       * @param {HTMLElement} clickedEmoji - .
+       */
+    console.log(clickedEmoji);
+    if (this.picked) {
```
Стоит уменьшить вложенность кода, тогда будет более читабельно.
Для этого нужно развернуть этот if:
```
if (!this.picked) {
    // do something if there is no picked emoji
    
    return; // важно именно закончить выполнение функции
}

// do anything else
```

gohabereg commented on main.js:
```diff
@@ -1,57 +1,142 @@
+       */
+    console.log(clickedEmoji);
+    if (this.picked) {
+      if (this.picked != clickedEmoji) {
```
С этим if можно аналогично поступить

gohabereg replied:
Если оба if развернуть, то станет гораздо лучше

sirpravdi commented on main.js:
```diff
@@ -1,57 +1,142 @@
-      this.parentElement.classList.add('reactions-wrapper__emoji_picked');
-      localStorage.setItem('index' + this.parentElement.dataset.reactionsCounter, parseInt(localStorage.getItem('index' + this.parentElement.dataset.reactionsCounter)) + 1);
-      wrap.querySelector('[data-reactions-index="index' + this.parentElement.dataset.reactionsCounter + '"]').innerText = localStorage.getItem('index' + this.parentElement.dataset.reactionsCounter);
+    const index = this.createElement('span', null, {innerText: this.getCounter(storageKey)});
```
Этот ключ меняется, я не могу его сохранить в перменную

gohabereg replied:
Он меняется на 0, и только если его не было в ls до этого. Код, который я предложил сверху, учитывает это

sirpravdi committed:
Update main.js

getCounter update to do parseInt inside of the method
changed some variables names
reactionClicked improved

gohabereg commented on main.js:
```diff
@@ -1,57 +1,144 @@
+    const storageKey = 'reactionIndex' + i;
+
+    label.dataset.index = i;
+    const input = this.createElement('input', null, {type: 'radio', name: 'poll'});
```
Думаю, input можно выпилить, ты его используешь только, чтобы клик повесить. А сам клик повесить на обертку.

sirpravdi replied:
и label'ы переделать в просто div'ы?

gohabereg commented on main.js:
```diff
@@ -1,57 +1,144 @@
+      
+      this.picked.classList.remove('reactions-wrapper__emoji--picked');
+      this.setCounter(pickedStorageKey, this.getCounter(pickedStorageKey) - 1);
+      this.picked.nextSibling.innerText = this.getCounter(pickedStorageKey);
```
Поиск по nextSibling лучше убрать, так как в коде не понятно, что это за элемент. Здесь как раз можно сделать `this.picked.querySelector('reactions-wrapper__counter')`. Тогда станет понятно, что за элемент ты ищешь.

gohabereg replied:
И во всех аналогичных случаях тоже лучше убрать nextSibling.

sirpravdi replied:
Но this.picked не содержит в себе span, да и reactions-wrapper__counter (это его родитель)
Я могу тут обратиться разве что через this.picked.parentElement.querySelector('span')

sirpravdi replied:
Ну или добавить к span класс и обращаться по имени класса

gohabereg commented on main.js:
```diff
@@ -1,57 +1,144 @@
+      this.picked.nextSibling.innerText = this.getCounter(pickedStorageKey);
+      clickedEmoji.classList.add('reactions-wrapper__emoji--picked');
+      this.setCounter(clickedStorageKey, this.getCounter(clickedStorageKey) + 1);
+      clickedEmoji.nextSibling.innerText = this.getCounter(clickedStorageKey);
```
Тоже несколько обращений к одному и тому же ключу в ls.
Здесь можно сделать так
```
const votes = this.getCounter(clickedStorageKey) + 1;
this.setCounter(clickedStorageKey, votes);

const counter = clickedEmoji.querySelector('reactions-wrapper__counter');
counter.innerHTML = votes;
```

gohabereg replied:
Тут чуть повыше и пониже в коде такие же ситуации, там тоже нужно поменять

gohabereg commented on main.js:
```diff
@@ -1,57 +1,144 @@
+  }
+
+  /** processing click on emoji */
+  reactionClicked(clickedEmoji) {
```
Добавь в этом методе пустых строк между кодом. Старайся, чтобы не было больше 3 строчек подряд, нужно разделять их логически

gohabereg commented on main.js:
```diff
@@ -1,57 +1,144 @@
+  /** processing click on emoji */
+  reactionClicked(clickedEmoji) {
+   /**
+       * processing click on emoji
```
Комменты к функции должны быть над функцией. Проверь везде по файлу

gohabereg commented on main.js:
```diff
@@ -1,57 +1,144 @@
+       * processing click on emoji
+       * @param {HTMLElement} clickedEmoji - .
+       */
+    if(!this.picked){
```
Теперь можно задокументировать каждый кейс.
Например тут 
```
/** If there is no previous picked reaction */
```

В следующем 
```
/** If clicked reaction is not equal to previous one */
```
 И в последнем тоже

sirpravdi committed:
Update main.css

sirpravdi committed:
Update main.js

Fix comments 
Fix code repetitions
Replace nextSibling with querySelector
Remove input as unnecessary

gohabereg commented on main.js:
```diff
@@ -1,57 +1,146 @@
+    const storageKey = 'reactionIndex' + i;
+
+    emoji.dataset.index = i;
+    emoji.addEventListener('click', click => this.reactionClicked(emoji));
```
Тут на самом деле можно передавать сразу i. Тогда можно вообще не записывать ничего в data-атрибуты

sirpravdi commented:
.
